### PR TITLE
Remove "remove in <version> comments"

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Remove the "remove in <version>" comments in feature gates. With compatibility version, features may only be removed after 3 releases post GA.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/triage accepted
/sig architecture
/assign @jpbetz

### CODE REVIEW

**Comment by carlory** [comment]
> Remove the "remove in " comments in feature gates. With compatibility version, features may only be removed after 3 releases post GA.

Do we need to change the deprecation-policy https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation

image

File: pkg/features/versioned_kube_features.go
Comment (liggitt) [inline pkg/features/versioned_kube_features.go:1]: I don't agree with removing the existing `drop in version <version>` comments

~all of the gates that locked to GA prior to 1.32 already had the relevant code checks removed, so we cannot actually emulate their behavior prior to locking true any more, and we should still drop the gates completely on the pre-existing schedule

if we want to ensure that *newly* graduating gates do not drop their conditional code checks for N releases, that *could* be ok, and *those* gates should not be dropped until the N releases are up, but should still be dropped eventually, and have a comment to that effect

Comment (liggitt) [reply pkg/features/versioned_kube_features.go:1]: /hold

**Comment by Jefftree** [comment]
/close

Per https://github.com/kubernetes/kubernetes/pull/127983#discussion_r1796848575, respecting the removal policy for feature graduated prior to 1.32.

**Pull request closed without merging**